### ISSUE TITLE
[isTypeReadonly] options parameter should be optional

### ISSUE THREAD

**Issue by RebeccaStevens** [typescript-eslint/typescript-eslint#4410]
- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [read the FAQ](https://github.com/typescript-eslint/typescript-eslint/blob/master/docs/getting-started/linting/FAQ.md) and my problem is not listed.

**Repro**

```TS
isTypeReadonly(checker, type)
```

**Expected Result**

A boolean.

**Actual Result**

Error, 3 params required.

**Additional Info**

https://github.com/typescript-eslint/typescript-eslint/blob/97c0e8606705dc0b4ef8b4ee5206111276c8a170/packages/type-utils/src/isTypeReadonly.ts#L246

**Versions**

| package                         | version |
| ------------------------------- | ------- |
| `@typescript-eslint/type-utils` | `5.9.0` |
| `@typescript-eslint/type-utils` | `5.9.0` |
| `TypeScript`                    | `4.5.4` |
| `node`                          | `17.1.0` |

Also note the error in the template in **Versions**: `@typescript-eslint/type-utils`'s version is requested twice. I assume there is supposed to be only one or one is supposed to be `@typescript-eslint/experimental-utils`
